### PR TITLE
Add nGraph visitor tests for reduction operations

### PR DESCRIPTION
**Pull request by ggalieroc**
Details:
Added typed case tests for nGraph visitor tests for reduction operations

- ReduceL1
- ReduceL2
- ReduceLogicalAnd
- ReduceLogicalOr
- ReduceMax
- ReduceMean
- ReduceMin
- ReduceProd
- ReduceSum

Tickets: 53982

### CODE REVIEW

File: ngraph/test/visitors/op/reduce_ops.cpp [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+using Types = ::testing::Types<
+    op::v1::ReduceLogicalAnd,
+    op::v1::ReduceLogicalOr,
+    op::v1::ReduceMax,
+    op::v1::ReduceMean,
+    op::v1::ReduceMin,
+    op::v1::ReduceProd,
+    op::v1::ReduceSum>;
+
+INSTANTIATE_TYPED_TEST_CASE_P(attributes, ReduceOpsAttrTest, Types);
```
Comment (pelszkow) [inline ngraph/test/visitors/op/reduce_ops.cpp:57]:
```suggestion
INSTANTIATE_TYPED_TEST_CASE_P(attributes, ReduceOpsAttrTest, Types);

```

File: ngraph/test/visitors/op/reduce_ops.cpp [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    NodeBuilder::get_ops().register_factory<TypeParam>();
+    auto data = make_shared<op::Parameter>(in_et, in_shape);
+    auto reduction_axes = make_shared<op::Parameter>(axes_et, axes_shape);
+    auto reduce_op = make_shared<TypeParam>(data, reduction_axes, keep_dims);
+
+    NodeBuilder builder(reduce_op);
+    auto g_reduce_op = as_type_ptr<TypeParam>(builder.create());
+    EXPECT_EQ(g_reduce_op->get_keep_dims(), reduce_op->get_keep_dims());
+}
```
Comment (pelszkow) [inline ngraph/test/visitors/op/reduce_ops.cpp:42]: IMHO we should add check for number of attributes in all `attribues` test. This fore developer to update test when new attributes will be add to op.
Suggest to add similar check:
https://github.com/openvinotoolkit/openvino/blob/master/ngraph/test/visitors/op/sqrt.cpp#L26-L28

**Review by pelszkow — APPROVED** [review]

**Review by blesniewski — APPROVED** [review]

**Comment by ggalieroc** [comment]
@ilyachur could you please have a look to this PR?

File: ngraph/test/visitors/op/reduce_ops.cpp [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    EXPECT_EQ(builder.get_value_map_size(), expected_attr_count);
+    auto g_reduce_op = as_type_ptr<TypeParam>(builder.create());
+    EXPECT_EQ(g_reduce_op->get_keep_dims(), reduce_op->get_keep_dims());
+}
+
+REGISTER_TYPED_TEST_CASE_P(
+    ReduceOpsAttrTest,
+    reduce_ops);
+
+using Types = ::testing::Types<
```
Comment (ilyachur) [inline ngraph/test/visitors/op/reduce_ops.cpp:50]: I propose to use the same approach as for type_prop tests: https://github.com/openvinotoolkit/openvino/blob/master/ngraph/test/type_prop/minimum.cpp

Comment (ggalieroc) [reply ngraph/test/visitors/op/reduce_ops.cpp]: Done

**Review by ilyachur — APPROVED** [review]

**Pull request merged**